smartcosmos-framework-monitoring/src/main/java/net/smartcosmos/metrics/PrometheusEndpoint.java

```diff
@@ -0,0 +1,130 @@
+package net.smartcosmos.metrics;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import io.prometheus.client.exporter.MetricsServlet;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.endpoint.Endpoint;
+import org.springframework.boot.actuate.endpoint.mvc.HypermediaDisabled;
+import org.springframework.boot.actuate.endpoint.mvc.MvcEndpoint;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.context.ServletContextAware;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.ServletWrappingController;
+import org.springframework.web.util.UrlPathHelper;
+
+/**
+ * Wraps the prometheus client MetricsServlet.
+ */
+@HypermediaDisabled
+public class PrometheusEndpoint implements MvcEndpoint, InitializingBean,
+                                           ApplicationContextAware, ServletContextAware,
+                                           EnvironmentAware {
+
+    private Environment environment;
```

Comment:
There is an `@Override public void setEnvironment()` method required by one of the interfaces. The field actually isn't used.